Refactor/v0.10

jhughes50 opened the pull request:
Update version of ouster_decoder that works with ouster sdk v0.10. It has been tested with ROS bags, the gen 6 and gen 0 lidars. I also organized the code base by putting all class and struct definition in their own header files in `include` and separated out the main function for each node. Finally, I updated documentation with a brief description of everything in the header files and an updated readme. I did create a new launch file to launch everything at once called `ouster.launch` and set the default parameter values to what we use. 

The logic has remained the same, I only made changes to how we get information from the ouster client and changed a few variable types to correspond with that.

versatran01 commented:
The repo contains a .clang-format file. Could you format your PR with that?

fcladera commented on launch/driver.launch:
```diff
@@ -1,17 +1,17 @@
-    <arg name="lidar_port" default="0" doc="port to which the sensor should send lidar data"/>
-    <arg name="imu_port" default="0" doc="port to which the sensor should send imu data"/>
-    <arg name="replay" default="true" doc="do not connect to a sensor; expect /os_node/{lidar,imu}_packets from replay"/>
+    <arg name="sensor_hostname" default="192.168.100.12" doc="hostname or IP in dotted decimal form of the sensor"/>
```
I'd suggest not changing these defaults, as this depends on the user.

fcladera commented on README.md:
```diff
@@ -1,8 +1,6 @@
-This decoder is intended to extend the ouster_ros package from https://github.com/ouster-lidar/ouster_example
-
-It has very low latency (<0.2ms) compared to ouster_example (>3ms), tested on Intel i7 11th gen cpu.
+This decoder is intended as an alternative the [ouster-ros](https://github.com/ouster-lidar/ouster-ros) decoder. It publishes things like LidarScans in a different formats which may be better for things like Lidar Odometry, while things like point clouds, signal, range and IMU remain in the same format. It also has very low latency (<0.2ms) compared to ouster_example (>3ms), tested on Intel i7 11 ...
```
@jhughes50 put together a few tests and I am stealing his slide here:
![image](https://github.com/KumarRobotics/ouster_decoder/assets/4083786/faaad5b5-f879-4349-b9fc-95d9f9d4aa85)

jhughes50 commented:
I updated the readme to reflect the new latency. I removed the sensor_hostname and udp_dest defaults and kept the port defaults, but can remove them if we don't want them in a launch file. I also changed the formatting to match throughout.
